habitat-lab/habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -650,9 +667,32 @@ def previous_step_collided(self):
             bool: True if the previous step resulted in a collision, false otherwise
 
         Warning:
-            This feild is only updated when :meth:`step`, :meth:`reset`, or :meth:`get_observations_at` are
-            called.  It does not update when the agent is moved to a new loction.  Furthermore, it
+            This field is only updated when :meth:`step`, :meth:`reset`, or :meth:`get_observations_at` are
+            called.  It does not update when the agent is moved to a new location.  Furthermore, it
             will _always_ be false after :meth:`reset` or :meth:`get_observations_at` as neither of those
             result in an action (step) being taken.
         """
         return self._prev_sim_obs.get("collided", False)
+
+    def add_keyframe_to_observations(self, observations):
+        r"""Adds an item to observations that contains the latest gfx-replay keyframe.
+        This is used to communicate the state of concurrent simulators to the batch renderer between processes.
+
+        :param observations: Original observations upon which the keyframe is added.
+        """
+        if not self.config.enable_batch_renderer:
```

Comment:
Based on how you're calling this, can we instead assert on this?

If we need to leave this code as-is, I argue for a function rename: I don't like functions called `do_x` that only do X conditionally. Sometimes I call them `try_do_x` or `check_do_x` to convey the uncertainty.